### ISSUE TITLE
iotune fails to detect XFS when evaluation-directory does not point to a mount point

### ISSUE THREAD

**Issue by ultrabug** [scylladb/scylla#3314]
*Installation details*
Scylla version (or git commit hash): 2.1.0

Context: commitlog and data directories are hosted on a single partition of a NVMe disk.

scylla.yaml shows:

```
data_file_directories:
    - /data/scylla01/data

commitlog_directory: /data/scylla01/commitlog
```

and partition is mounted as:

```
/dev/nvme0n1p1 on /data/scylla01 type xfs (rw,noatime,attr2,inode64,noquota)
```

then you get the following error:

```
ERROR:root:/data/scylla01/data did not pass validation tests, it may not be on XFS and/or has limited disk space.
This is a non-supported setup, and performance is expected to be very bad.
For better performance, placing your data on XFS-formatted directories is required.
To override this error, enable developer mode as follow:
sudo /usr/lib/scylla/scylla_dev_mode_setup --developer-mode
```

This means that iotune disk detection should be improved to detect the top level mount point when checking for XFS.

I'll try and propose a patch.

**Comment by ultrabug** [comment]
Actually, the error is not about XFS detection but is due to the fact that the directory was not existing

Running iotune by hand shows the real error:
```
terminate called after throwing an instance of 'std::system_error'
  what():  open: No such file or directory
Aborted
```

So the fix is simpler as to check for the `evaluation-directory` before running the command.

**Comment by ultrabug** [comment]
Ok, it has been fixed on master by Takuya ASADA via 5f835be3aa890a44714922883eaef48076895fd3
